charms/istio-gateway/src/manifest.yaml

```diff
@@ -0,0 +1,324 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app: istio-{{ kind }}gateway
+    install.operator.istio.io/owning-resource: unknown
+    istio: {{ kind }}gateway
+    istio.io/rev: default
+    operator.istio.io/component: "{{ kind|capitalize }}Gateways"
+    release: istio
+  name: istio-{{ kind }}gateway-service-account
+  namespace: {{ namespace }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: istio-{{ kind }}gateway
+    install.operator.istio.io/owning-resource: unknown
+    istio: {{ kind }}gateway
+    istio.io/rev: default
+    operator.istio.io/component: {{ kind|capitalize }}Gateways
+    release: istio
+  name: istio-{{ kind }}gateway
+  namespace: {{ namespace }}
+spec:
+  selector:
+    matchLabels:
+      app: istio-{{ kind }}gateway
+      istio: {{ kind }}gateway
+  strategy:
+    rollingUpdate:
+      maxSurge: 100%
+      maxUnavailable: 25%
+  template:
+    metadata:
+      annotations:
+        prometheus.io/path: /stats/prometheus
+        prometheus.io/port: "15020"
+        prometheus.io/scrape: "true"
+        sidecar.istio.io/inject: "false"
+      labels:
+        app: istio-{{ kind }}gateway
+        chart: gateways
+        heritage: Tiller
+        install.operator.istio.io/owning-resource: unknown
+        istio: {{ kind }}gateway
+        istio.io/rev: default
+        operator.istio.io/component: {{ kind|capitalize }}Gateways
+        release: istio
+        service.istio.io/canonical-name: istio-{{ kind }}gateway
+        service.istio.io/canonical-revision: latest
+        sidecar.istio.io/inject: "false"
+    spec:
+      affinity:
+        nodeAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - preference:
+                matchExpressions:
+                  - key: kubernetes.io/arch
+                    operator: In
+                    values:
+                      - amd64
+              weight: 2
+            - preference:
+                matchExpressions:
+                  - key: kubernetes.io/arch
+                    operator: In
+                    values:
+                      - ppc64le
+              weight: 2
+            - preference:
+                matchExpressions:
+                  - key: kubernetes.io/arch
+                    operator: In
+                    values:
+                      - s390x
+              weight: 2
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: kubernetes.io/arch
+                    operator: In
+                    values:
+                      - amd64
+                      - ppc64le
+                      - s390x
```

Comment:
Are we actuallly supporting these architectures?

Reply:
Charmed Kubernetes supports those architectures to my knowledge, and is interested in using these charms for Ingress. Also, even if we stick with x86 only, I think it reduces a lot of maintenance burden to minimize diffs between upstream YAML and ours.